### PR TITLE
PLAT-629: replace reference builder with rms lazy references

### PR DESCRIPTION
**Pull request by bigbes**
later we can integrate that machanism into reference builder, if needed

### CODE REVIEW

File: ledger-core/virtual/vnlmn/register.go [hunk 1]
```diff
@@ -137,10 +159,10 @@ func (s *RegisterRecordBuilder) getOutboundRecord() *rms.ROutboundRequest {
 
 	// first outgoing of incoming should be branched from
 	prevRef := s.lastBranchRef
-	if prevRef.IsEmpty() {
+	if prevRef == nil {
```
Comment (cyraxred) [inline ledger-core/virtual/vnlmn/register.go:162]: I'd recommend use of reference.IsEmpty

File: ledger-core/virtual/vnlmn/register_waitsafe.go [hunk 2]
```diff
@@ -58,7 +58,7 @@ func (s *SMWaitSafeResponse) bargeInLRegisterResponseHandler(param interface{})
 	}
 
 	return func(ctx smachine.BargeInContext) smachine.StateUpdate {
-		if res.AnticipatedRef != s.ExpectedKey.AnticipatedRef || res.Flags != s.ExpectedKey.RequiredFlag {
+		if !res.AnticipatedRef.Equal(&s.ExpectedKey.AnticipatedRef) || res.Flags != s.ExpectedKey.RequiredFlag {
```
Comment (cyraxred) [inline ledger-core/virtual/vnlmn/register_waitsafe.go:61]: if AnticipatedRef is reference.Global then there is no need for "&" here

File: ledger-core/virtual/vnlmn/register_context.go [hunk 3]
```diff
@@ -8,9 +8,26 @@ package vnlmn
 import (
 	"github.com/insolar/assured-ledger/ledger-core/conveyor/smachine"
 	"github.com/insolar/assured-ledger/ledger-core/reference"
+	"github.com/insolar/assured-ledger/ledger-core/rms/rmsbox"
 	"github.com/insolar/assured-ledger/ledger-core/vanilla/throw"
 )
 
+type rawValue struct {
```
Comment (cyraxred) [inline ledger-core/virtual/vnlmn/register_context.go:15]: better to add/move it to rms package pls

**Review by cyraxred — APPROVED** [review]

File: ledger-core/virtual/vnlmn/register_context.go (hunk 3 above)
Comment (bigbes) [reply ledger-core/virtual/vnlmn/register_context.go]: Renamed to `rawValueProvider` and moved to `rmsbox`.

**Review by MikhailZhukov — APPROVED** [review]

**Review by cyraxred — APPROVED** [review]

**Pull request merged**